routes/parking.js

```diff
@@ -14,26 +66,68 @@ router.post('/create-parking', async (req, res) => {
     return res.status(400).send('Missing parameter')
   }
 
-  try {
-    const parking = new Parking({
-      name: name,
-      number: number,
-      rate: rate,
-      address: address
-    })
-    QRCode.toDataURL(parking.id, { width: 300 }, function (err, url) {
-      if (err) {
-        return res.status(500).send("failed to create qrcode");
-      }
-      // If qrcode generated successfully, we save the document for the space.
-      // console.log(url);
-      parking.qrCodeUrl = url;
-    })
+  const parking = new Parking({
+    name: name,
+    number: number,
+    rate: rate,
+    address: address,
+
+  })
+  QRCode.toDataURL(parking.id, { width: 300 }, function (err, url) {
+    if (err) {
+      return res.status(500).send("failed to create qrcode");
+    }
+    // If qrcode generated successfully, we save the document for the space.
+    parking.qrCodeUrl = url;
     parking.save()
-  } catch (err) {
-    console.log(err)
-    res.status(400).send(`create parking failed ${err}`)
+      .then(parking => {
+        addToCache(parking.toObject())
+        res.sendStatus(200)
+      })
+      .catch(err => res.status(400).send(`create parking failed ${err}`))
+  })
+})
+
+router.get('/all', async (req, res) => {
+  res.send(cachedParkingLots);
+})
+
+// router.put('/session', async (req, res) => {
+
+//   // emits map with key: parking-id, value: session object
+//   io.sockets.emit("session-map", sessionMap)
+// })
+
+// io.on("connection", socket => {
+//   console.log("New client connected: " + socket.id);
+// })
+
+router.delete('/:parkingId', async (req, res) => {
+  Parking.findByIdAndDelete(req.params.parkingId).exec()
+    .then(parking => deleteFromCache(req.params.parkingId))
+    .catch(err => res.status(400).send(`delete parking failed ${err}`))
+})
+
+router.put('/:parkingId', async (req, res) => {
+  const { name, number, rate, address, qrCodeUrl } = req.body
+
+  if (!name || !number || !rate || !address || !qrCodeUrl) {
+    return res.status(400).send('Missing parameter')
   }
+
+  const parkingLot = { name, number, rate, address, qrCodeUrl }
+
+  console.log(parkingLot)
+  Parking.findOneAndReplace({ "_id": ObjectId(req.params.parkingId) }, parkingLot, function (err, parkingLot) {
```

Comment:
Use `findOneAndUpdate` instead of `findOneAndReplace`

Reply:
or FindByIdAndUpdate would be easier

Reply:
We could do this, but I wanted to keep it as general as possible, as we're not sure which field in the parking lot the admin will choose to update (could be just the name, could be the name and address, etc) so i was thinking to have the react app just send the entire parking lot object over.